### PR TITLE
additional metrics (added path length)

### PR DESCRIPTION
**Pull request by rnfinnegan**

### CODE REVIEW

File: platipy/imaging/label/comparison.py [hunk 1]
```diff
@@ -293,3 +293,94 @@ def compute_metric_hd(label_a, label_b, auto_crop=True):
     hausdorff_distance_value = hausdorff_distance.GetHausdorffDistance()
 
     return hausdorff_distance_value
+
+
+def compute_metric_total_apl(label_ref, label_test, distance_threshold_mm=0):
```
Comment (pchlap) [inline platipy/imaging/label/comparison.py:298]: Do you think 0 is a good default value here? For the surface dsc I chose 3mm (that seems to be a common value). In practice you mentioned IoV MDA would be the correct value to choose... but I wonder if 3mm is a reasonable estimate for "most" cases?

File: platipy/imaging/label/comparison.py [hunk 2]
```diff
@@ -293,3 +293,94 @@ def compute_metric_hd(label_a, label_b, auto_crop=True):
...
+            label_ref_contour = sitk.BinaryDilate(label_ref_contour, (distance_threshold,distance_threshold))
+
+        # mask out the locations in agreement
+        added_path = sitk.MaskNegated(label_test_contour, label_ref_contour)
+
+        # add up the voxels on the added path
+        added_path_length = sitk.GetArrayViewFromImage(added_path).sum()
+        added_path_length_list.append(added_path_length)
+
+    mean_added_path_length = np.mean(added_path_length_list)
```
Comment (pchlap) [inline platipy/imaging/label/comparison.py:381]: Are these two functions identical expect for this line? Could be combined into one function? Or at least pull the common bits out into one function they both call?

**Review by pchlap — COMMENTED** [review]
Thanks Rob, left a couple of comments to consider?

File: platipy/imaging/label/comparison.py (hunk 1 above)
Comment (rnfinnegan) [reply platipy/imaging/label/comparison.py:298]: Good thought. I'm a bit conflicted, as the APL was originally designed to compare autocontours to edited autocontours - i.e. most of the contour would be identical (so distance = 0) and in such cases one would want to measure any change made. If you chose 3mm for the surface dice then I think being consistent and using 3mm here would be good!

File: platipy/imaging/label/comparison.py (hunk 2 above)
Comment (rnfinnegan) [reply platipy/imaging/label/comparison.py:381]: yeah they are, let's definitely do that!

Comment (rnfinnegan) [reply platipy/imaging/label/comparison.py:390]: Hey Phil,

Great comments, thanks! I've refactored the code, and also set the default parameter to 3mm.

**Comment by rnfinnegan** [comment]
Hey Phil, great comments, thanks! I've refactored the code, and also set the default parameter to 3mm.

File: platipy/imaging/label/comparison.py [hunk 3]
```diff
@@ -356,38 +351,37 @@ def compute_metric_mean_apl(label_ref, label_test, distance_threshold_mm=0):
...
-
-        if (
-            sitk.GetArrayViewFromImage(label_ref)[i].sum()
-            + sitk.GetArrayViewFromImage(label_ref)[i].sum()
-        ) == 0:
-            continue
 
-        label_ref_contour = sitk.LabelContour(label_ref[:, :, i])
-        label_test_contour = sitk.LabelContour(label_test[:, :, i])
+def compute_metric_mean_apl(label_ref, label_test, distance_threshold_mm=0):
```
Comment (pchlap) [inline platipy/imaging/label/comparison.py:366]: Change to 3mm

**Review by pchlap — APPROVED** [review]

**Pull request merged**